File: app/src/main/java/com/example/speedrun/injection/components/ActivityComponent.java
package com.example.speedrun.injection.components;

import com.example.speedrun.ui.main.MainActivity;
import com.example.speedrun.injection.PerActivity;
import com.example.speedrun.injection.modules.ActivityModule;
import com.example.speedrun.ui.base.BaseActivity;
import com.example.speedrun.ui.splash.SplashActivity;

import dagger.Component;

@PerActivity
@Component(
        dependencies = {ApplicationComponent.class},
        modules = {ActivityModule.class})
public interface ActivityComponent {

    void inject(BaseActivity activity);

    void inject(MainActivity activity);
    void inject(SplashActivity activity);

    //Context activityContext();
}

File: app/src/main/java/com/example/speedrun/injection/components/ActivityComponent.kt
package com.example.speedrun.injection.components

import com.example.speedrun.injection.modules.ActivityModule
import com.example.speedrun.injection.scopes.ActivityScope
import com.example.speedrun.ui.base.BaseActivity
import com.example.speedrun.ui.game.GameDetailsActivity
import com.example.speedrun.ui.main.MainActivity
import com.example.speedrun.ui.run.RunDetailsActivity
import com.example.speedrun.ui.splash.SplashActivity
import com.example.speedrun.ui.user.UserProfileActivity
import dagger.Component

@ActivityScope
@Component(dependencies = [ApplicationComponent::class], modules = [ActivityModule::class])
interface ActivityComponent {
    fun inject(activity: BaseActivity)

    fun inject(activity: MainActivity)
    fun inject(activity: SplashActivity)
    fun inject(activity: UserProfileActivity)
    fun inject(activity: GameDetailsActivity)
    fun inject(activity: RunDetailsActivity)
    //Context activityContext();
}